### PR TITLE
add kubeadm and floating tags for different kubernetes versions

### PR DESCRIPTION
**Pull request by Chris-256**
- add kubeadm, cfssl and cfssljson
- add a travis cronjob building the image for the latest kube version
- add a travis job rebuilding the image for the last 4 minor versions of kubernetes (triggering on push events)

Tagging scheme:
<1.12.1-1.2.2> = specific kube and repo version
<1.12.1-latest> = specific kube and latest repo version
\<latest> = latest kube and latest repo version

### CODE REVIEW

File: bin/build [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+#
+# Builds docker images for different minor versions of kubernetes
+#
+
+set -o errexit
+set -o pipefail
```
Comment (tonyfahrion) [inline bin/build]: please add `set -o nounset` as well

File: Dockerfile [hunk 2]
```diff
@@ -22,10 +24,15 @@ RUN pip install --upgrade pip \
...
-RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl; \
-    mv kubectl /usr/bin/kubectl; \
-    chmod +x /usr/bin/kubectl
+# Install cfssl and cfssljson
+RUN curl -sSL https://pkg.cfssl.org/R1.2/cfssl_linux-amd64 > /usr/bin/cfssl \
+ && curl -sSL https://pkg.cfssl.org/R1.2/cfssljson_linux-amd64 > /usr/bin/cfssljson \
+ && chmod +x /usr/bin/cfssl /usr/bin/cfssljson
+
+# Install kubectl and kubeadm
+RUN curl -L https://storage.googleapis.com/kubernetes-release/release/${KUBE_VERSION}/bin/linux/amd64/kubectl > /usr/bin/kubectl \
```
Comment (tonyfahrion) [inline Dockerfile]: you can use `curl -L --remote-name <url>` here, so you can save the stdout redirection

File: Dockerfile [hunk 3]
```diff
@@ -22,10 +24,15 @@ RUN pip install --upgrade pip \
...
+ && curl -L https://storage.googleapis.com/kubernetes-release/release/${KUBE_VERSION}/bin/linux/amd64/kubeadm > /usr/bin/kubeadm \
```
Comment (tonyfahrion) [inline Dockerfile]: and I would like to have the URL possible to be given as an `ARG`. You could default it, if you want 🙂 

Something like
```Dockerfile
ARG KUBE_BINARY_URL="https://storage.googleapis.com/kubernetes-release/release/${KUBE_VERSION}/bin/linux/amd64"
RUN curl -L --remote-name ${KUBE_BINARY_URL}/kubectl
```

What do you think about that? Is it too much, or does it make the code more readable/maintainable (that would be my hope)?

File: bin/build [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+print_usage() {
+    printf "\nUsage: $0 <latest|last4minors> <TAG>\n\n"
+    exit 1
+}
+
+[[ -z $TAG ]] && print_usage
+
+case $1 in
+    latest)
+    	build_minor $MINOR_VERSION
```
Comment (tonyfahrion) [inline bin/build]: please call it `build_image` and please differentiate between build, test and push (different commands each)

I know that's how it's done before; but I think we can improve here

File: bin/image [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	
+case $1 in
+	
+    build)
+    	build_images $versions
+		;;
+    test)
+    	test_images $versions
+		;;
+	push)
```
Comment (tonyfahrion) [inline bin/image]: please fix indentation here; after that we are ready for the CHANGELOG.md and bumpversion 🙂 👍

File: .travis.yml [hunk 6]
```diff
@@ -5,7 +5,7 @@ jobs:
   include:
     
     - name: Build and push docker image with the newest kube-version
-      if: type = cron
+      if: type = cron AND branch = master
```
Comment (tonyfahrion) [inline .travis.yml]: hmm, isn't this an `OR` condition? I would guess that we need to specify the branch in travis-ci on which the cron will run on

Comment (Chris-256) [reply .travis.yml]: The `branch = master` is not strictly necessary if we specify the branch, but with `OR` the job would also run on push events which is not intended.

**Comment by tonyfahrion** [comment]
@Chris-256 Thank you for the PR! Now you can push the tag 🙂

**Pull request merged**